discovery-provider/src/models/models.py

```diff
@@ -32,6 +32,24 @@
 logger = logging.getLogger(__name__)
 
 
+# Autogenerate __repr__ for models
+class RepresentableMixin:
```

Comment:
Nice! Maybe the docstring should go inside for intellisense? I'm not sure intellisense picks up the inline comments